src/vm_pages.rs

```diff
@@ -1334,6 +1463,79 @@ impl<'a, T: GuestStagePagingMode> FinalizedVmPages<'a, T> {
         Ok(())
     }
 
+    /// Converts the specified memory region from shared to confidential. Returns the TLB version
+    /// at which the conversion will be completed.
+    pub fn unshare_mem_region(&self, page_addr: GuestPageAddr, len: u64) -> Result<()> {
```

Comment:
the `*unshare_mem_region` methods look pretty similar to the `*share_mem_region` ones except the region types are flipped. is it possible to deduplicate these a bit?